modules/util.py

```diff
@@ -124,39 +127,78 @@ class PomUtils:
     def __init__(self, driver: Firefox):
         self.driver = driver
 
-    def get_shadow_content(self, element: WebElement) -> list[WebElement]:
+    def get_shadow_content(
+        self, element: WebElement
+    ) -> list[Union[WebElement, ShadowRoot]]:
         """
         Given a WebElement, return the shadow DOM root or roots attached to it. Returns a list.
         """
-        try:
-            shadow_root = element.shadow_root
-            return [shadow_root]
-        except InvalidArgumentException:
+        logging.info(f"Getting shadow nodes from root {element}")
+
+        def shadow_from_script():
             shadow_children = self.driver.execute_script(
                 "return arguments[0].shadowRoot.children", element
             )
             if len(shadow_children) and any(shadow_children):
```

Comment:
I don't quite understand this logic. Could you explain why ```any(shadow_children)``` is needed?

Reply:
When Selenium parses the results of the Javascript, it could be that any given child node is null, meaning we could have a list like `[None, None]` which I don't think is falsy. Let me check that...yep, that's considered `True` in python

Reply:
A list of None's is true lol, I love python. I guess a room full of no one is still a room!